src/org/labkey/filetransfer/view/StyleableActionButton.java

```diff
@@ -0,0 +1,551 @@
+package org.labkey.filetransfer.view;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.labkey.api.data.DataRegion;
+import org.labkey.api.data.RenderContext;
+import org.labkey.api.query.DetailsURL;
+import org.labkey.api.security.permissions.DeletePermission;
+import org.labkey.api.security.permissions.InsertPermission;
+import org.labkey.api.security.permissions.UpdatePermission;
+import org.labkey.api.util.Button;
+import org.labkey.api.util.MemTracker;
+import org.labkey.api.util.PageFlowUtil;
+import org.labkey.api.util.StringExpression;
+import org.labkey.api.util.StringExpressionFactory;
+import org.labkey.api.util.URLHelper;
+import org.labkey.api.view.ActionURL;
+import org.labkey.api.view.DisplayElement;
+import org.springframework.web.servlet.mvc.Controller;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by dwightman on 9/26/2017.
+ */
+public class StyleableActionButton  extends DisplayElement implements Cloneable
```

Comment:
Your reading is correct--we needed to add CSS classes, but couldn't use a regular Button because it didn't have content-aware methods like setRequiresSelection().  There's a comment in ActionButton saying it needs to have its inheritance rejiggered, which may be why we didn't touch it directly.  I'll let Dennis weigh in.

Reply:
That was my original idea, but iirc I couldn't make that work.  When we revisit it, I'll take another look.